tests/test_main.py

```diff
@@ -1,18 +1,31 @@
+from pathlib import Path
 from unittest.mock import patch
 
+import pytest
 from napari_micromanager.__main__ import main
+from pymmcore_plus import CMMCorePlus
 
 
-def test_cli_main() -> None:
+@pytest.mark.parametrize(
+    "argv",
+    [[""], ["", str(Path(__file__).parent / "test_config.cfg")], ["", "nonexistant"]],
+)
+def test_cli_main(argv: list) -> None:
     import napari
     from napari.qt import QtViewer
 
     with patch("napari.run") as mock_run:
         with patch("qtpy.QtWidgets.QMainWindow.show") as mock_show:
-            main()
+            with patch("napari.utils.notifications.show_warning") as mock_warning:
```

Comment:
It seems that this patch isn't actually patching (i.e. the warning still gets displayed) and it is never called.

Any idea what I'm doing wrong here?

Reply:
not immediately, but you know what... just use `warnings.warn()`, napari will intercept that anyway and show a warning in the UI (or at least it used to?)

Reply:
hmm it doesn't seem to capture it in the same way. maybe i need to change the level

Reply:
oh, it's possible that the napari warnings hook hasn't yet been invoked by this point?

Reply:
actually scratch that, now the show warnign is also not showing, so maybe just going away to quickly. it still shows up in teh terminal